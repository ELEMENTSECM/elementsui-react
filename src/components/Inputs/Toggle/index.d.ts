import * as React from 'react';

export interface ToggleProps {
	/**
	 * Toggle label
	 */
	label?: string;
	/**
	 * Toggle is disabled
	 */
	disabled?: boolean;
	/**
	 * Toggle description
	 */
	ariaDescribedBy?: string;
	/**
	 * Toggle checked
	 */
	defaultChecked?: boolean;
	/**
	 * User-defined styling
	 */
	styles?: (...args: any[]) => any;
}

declare const Toggle: React.SFC<ToggleProps>;

export default Toggle;
