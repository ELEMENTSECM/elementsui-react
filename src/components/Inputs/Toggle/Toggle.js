import * as React from 'react';
import PropTypes from 'prop-types';
import { Toggle as UIFabToggle } from 'office-ui-fabric-react/lib/Toggle';
import { styles } from './Toggle.styles';
import { classNamesFunction, customizable, styled } from 'office-ui-fabric-react/lib/Utilities';

/** Toggle example */
export function Toggle(props) {
	const { label, className, theme, styles, disabled = false } = props;
	const classNames = classNamesFunction()(styles, props);
	return (
		<UIFabToggle
			className={classNames.root}
			label={label}
			disabled={disabled}
			ariaDescribedBy={'descriptionID'}
			defaultChecked="false"
		/>
	);
}

Toggle.propTypes = {
	/** Toggle label */
	label: PropTypes.string,
	/** Toggle is disabled */
	disabled: PropTypes.bool,
	/** Toggle description */
	ariaDescribedBy: PropTypes.string,
	/** Toggle checked */
	defaultChecked: PropTypes.bool,
	/** User-defined styling */
	styles: PropTypes.func
};

export default styled(customizable('Toggle', ['theme'])(Toggle), styles);
