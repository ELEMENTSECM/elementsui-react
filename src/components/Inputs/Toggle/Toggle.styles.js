export const styles = props => {
	const { className, theme } = props;
	const { palette, semanticColors } = theme;

	return {
		root: [
			'ms-Toggle',
			{
				// place your styles here
			},
			className
		]
	};
};
