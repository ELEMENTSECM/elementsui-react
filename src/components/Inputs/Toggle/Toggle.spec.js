import React from 'react';
import renderer from 'react-test-renderer';
import Toggle from './Toggle';

describe('Toggle', () => {
	test('displays `Foo` as label', () => {
		const tree = renderer
			.create(<Toggle htmlId="defaultToggle" label="Foo"/>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
