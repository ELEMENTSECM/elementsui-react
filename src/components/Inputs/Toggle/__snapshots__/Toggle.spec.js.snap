// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle displays \`Foo\` as label 1`] = `
<div
  className="ms-Toggle is-checked is-enabled root-41"
>
  <label
    className="ms-Label ms-Toggle-label label-47"
    htmlFor="Toggle1"
  >
    Foo
  </label>
  <div
    className="ms-Toggle-innerContainer container-43"
  >
    <button
      aria-disabled={false}
      aria-label="Foo"
      aria-pressed={true}
      className="ms-Toggle-background pill-44"
      data-is-focusable={true}
      disabled={false}
      id="Toggle1"
      onChange={[Function]}
      onClick={[Function]}
      type="button"
    >
      <div
        className="ms-Toggle-thumb thumb-45"
      />
    </button>
  </div>
</div>
`;
